packages/amplify-appsync-simulator/tsconfig.json

```diff
@@ -5,9 +5,21 @@
     "esModuleInterop": true,
     "sourceMap": true,
     "outDir": "lib",
-    "lib": ["es2015", "es2016.array.include", "esnext.asynciterable", "dom"],
+    "composite": true,
+    "rootDir": "./src",
+    "lib": [
+      "es2015",
+      "es2016.array.include",
+      "esnext.asynciterable",
+      "dom"
+    ],
     "declaration": true,
     "skipLibCheck": true,
   },
-  "exclude": ["node_modules", "lib", "**/__mocks__/**/*", "**/__tests__/**/*"]
-}
+  "exclude": [
+    "node_modules",
+    "lib",
+    "**/__mocks__/**/*",
+    "**/__tests__/**/*"
```

Comment:
Don't we have to compile the ts files moved inside `__tests__/server` folder?

Reply:
This is an interesting question about how the TS files in __tests__ are compiled. Doing a bit of investigation, `jest` actually takes care of `TS->JS` transformation for the test files it discovers. We configure how to do this using the below config which you can find in [package.json]()
```
"transform": {
      "^.+\\.tsx?$": "ts-jest"
    }
```
This basically means for all the TS files, use `ts-jest`(built-in) to transform them to JS before executing the test.
So, each time you run the test, it's also inherently compiled by `jest`. We use this for several packages already. 

Resources: 
- https://jestjs.io/docs/configuration#transform-objectstring-pathtotransformer--pathtotransformer-object
- https://jestjs.io/docs/code-transformation